core/src/main/java/org/apache/iceberg/util/ThreadPools.java

```diff
@@ -61,6 +62,26 @@ public static ExecutorService getWorkerPool() {
     return WORKER_POOL;
   }
 
+  public static ExecutorService newWorkerPool(String namePrefix, Integer parallelism) {
+    return MoreExecutors.getExitingExecutorService(
+        (ThreadPoolExecutor) Executors.newFixedThreadPool(
+            Optional.ofNullable(parallelism).orElse(WORKER_THREAD_POOL_SIZE),
+            new ThreadFactoryBuilder()
+                .setDaemon(true)
+                .setNameFormat(namePrefix + "-%d")
+                .build()));
+  }
+
+  public static ExecutorService newKeyedWorkerPool(String key, String namePrefix, Integer parallelism) {
```

Comment:
if the intention is to reuse the job specific thread pool in Flink, then we do need the static cache as the same keyed pool may be requested from multiple code path. 

Is this a Flink only problem regarding classloader issue on thread pool? if so, maybe we can move the keyed cache into Flink module.

Reply:
Oh, so the job can share between the monitor and the sink? I don't really mind having two pools for that.

Reply:
@rdblue ,sorry, i don't get your point exactly. let me guess, what you really mean is sharing pools for all sources or all sinks, not for all sources and sinks?
To be clear, for example, if a job consists of:
Source: Iceberg A(parallelism: 3), Source:Iceberg B, Sink:Iceberg C, Sink: Iceberg D.
What's your favor?

1. share btw all parallelism of one operator, like 3 subtask for Iceberg A (it can be run in different slots in one TaskManager or different TaskManagers) ;
2. share btw all sources or all sinks, like sharing one for btw A and B, and another one for C and D; 
3. share btw all operators, like sharing btw A, B, C, and D; all subtasks in same TaskManager can share.

Reply:
What I was thinking was a pool per operator in a job, rather than a pool per job. That avoids the need to track thread pools by some key in static state. I think it is probably fine to have more pools since these are primarily for IO. Does that sound reasonable?

Reply:
So no any kind of sharing is needed mentioned [here#comment](https://github.com/apache/iceberg/pull/4177#discussion_r812199983), right? That makes sense to me.

Reply:
after reviewing the usage of the thread pools, I am also in favor of no sharing of thread pools so that we can avoid the static cache. None of the usage is on parallel tasks.
* source: split planning (running on jobmanager or the single-parallelism StreamingMonitorFunction)
* sink: single-parallelism committer

Reply:
@stevenzwu, for that use case, maybe we should follow up to this PR with one that allows you to configure a named threadpool? I think that's probably the use case that @yittg had in mind when he set up sharing.

Reply:
In addition to documentation change, we should also make sure this behavior change is captured in the release note of the next minor version release of 0.14.0.  @rdblue where do we track future release note?

Reply:
@stevenzwu, I added the "release notes" tag to this PR and added it to the 0.14.0 release milestone so we add this to release notes. If you want, you can add a comment with the suggested release notes at the end.

Reply:
Yeah, although I didn't think about sharing at the beginning. but some kind of sharing or global limit sounds good to me after some consideration. We can provide a reasonable solution next I think. 
Thanks, @rdblue and @stevenzwu .